File: 151002/Rusakovich/NewsDistributedSystem/publisher/src/main/java/by/rusakovich/publisher/label/model/Label.java
package by.rusakovich.publisher.label.model;

import by.rusakovich.publisher.generics.model.IEntity;
import by.rusakovich.publisher.news.model.NewsLabel;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "tbl_label")
public class Label implements IEntity<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 32)
    private String name;

    @OneToMany(mappedBy = "label")
    private List<NewsLabel> news;
}

File: 151001/Novik/DC/src/Publisher/Models/Entity/Label.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Publisher.Models.Entity;

public class Label
{
    [Key]
   
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    
    public long id { get; set; }
    
    [StringLength(32, MinimumLength = 2)]
    public String? name { get; set; }

    public ICollection<News> News { get; set; }= new List<News>();

    public ICollection<NewsLabel> NewsLabels { get; set; }= new List<NewsLabel>();
}